app/controllers/test_model_version_runs_controller.rb

```diff
@@ -34,15 +34,7 @@ def set_test_run
     @test_run = current_user.test_runs.find(params[:test_run_id])
   end
 
-  def perform_all_test_model_version_run_jobs
-    ApplicationRecord.transaction do
-      @test_model_version_run.lock!
-      break false if @test_model_version_run.performed?
-
-      jobs = Array.new(@test_run.calls) { TestModelVersionRunJob.new(@test_model_version_run.id) }
-      SolidQueue::Job.enqueue_all(jobs)
-
-      @test_model_version_run.update(performed: true)
-    end
+  def perform_test_model_version_run_jobs
+    PerformTestModelVersionRunJobs.new(@test_run.id, @test_model_version_run.id).call
```

Comment:
I don't like these extra queries being executed in the before_action hooks for all endpoints. We only need to check if the current user has access to the test_run with the given ids.

Reply:
skipped `set_test_model_version_run` hook for the `perform` action